src/rust/catnap/linux/transport.rs

```diff
@@ -368,39 +362,13 @@ impl SharedSocketData {
 }
 
 impl SharedCatnapTransport {
-    /// Create a new Linux-based network transport.
-    pub fn new(_config: &Config, mut runtime: SharedDemiRuntime) -> Self {
-        // Create epoll socket.
-        // Linux ignores the size argument to epoll, it just has to be more than 0.
-        let epoll_fd: RawFd = match unsafe { libc::epoll_create(10) } {
-            fd if fd >= 0 => fd.into(),
-            _ => {
-                let errno: libc::c_int = unsafe { *libc::__errno_location() };
-                panic!("could not create epoll socket: {:?}", errno);
-            },
-        };
-
-        // Set up background task for polling epoll API.
-        let yielder: Yielder = Yielder::new();
-        let background_task: YielderHandle = yielder.get_handle();
-        let me: Self = Self(SharedObject::new(CatnapTransport {
-            epoll_fd,
-            socket_table: Slab::<SharedSocketData>::new(),
-            background_task,
-        }));
-        let mut me2: Self = me.clone();
-        runtime
-            .insert_background_coroutine(
-                "catnap::transport::epoll",
-                Box::pin(async move { me2.poll(yielder).await }),
-            )
-            .expect("should be able to insert background coroutine");
-        me
-    }
-
     /// This function registers a handler for incoming and outgoing I/O on the socket. There should only be one of
     /// these per socket.
-    fn register_epoll(&mut self, sd: &SocketDescriptor, events: u32) -> Result<(), Fail> {
+    fn register_epoll(
+        &mut self,
+        sd: &<SharedCatnapTransport as NetworkTransport>::SocketDescriptor,
```

Comment:
Is it possible to move the `<SharedCatnapTransport as NetworkTransport>::SocketDescriptor` to the top of the file in a `use` expression?

Reply:
Sure, I didn't know that we could do that. Thanks!